### ISSUE TITLE
New post not getting published

### ISSUE THREAD

**Issue by notpratheek** [hmngwy/weblog.sh#29]
Hey

I tried to post a new post, but that doesn't seem to come up ! :sweat_smile: 
## 

First, I tried to login into my account (with my password)

```
$ ssh -2 -p 2222 pthk@weblog.sh
Reset key using:

  ssh undefined@weblog.sh key $(cat ~/.ssh/id_rsa.pub)

Connection to weblog.sh closed.                       
$ 
```

So, I went back to the main site (weblog.sh), and saw that you had made the adding of ssh key mandatory, so I created one, by running the command  `ssh-keygen -t rsa` after the key was generated (I created it with a passphrase),I tried to add it by running  `ssh -2 -p 2222 pthk@weblog.sh key $(cat ~/.ssh/id_rsa.pub)`, It first asked for my password, which I typed in, then it showed "adding key ..." followed by "key added to user".

So, now I tried to publish a post via

```
$ scp -P 2222 file_in_md.md pthk@weblog.sh:public/
```

It showed that the file successfully copied over, so I went to site (weblog.sh/~pthk) and only the old one was there, the new one didn't get published ! 

I also tried to run 

```
ssh -2 -p 2222 pthk@weblog.sh publish file_in_md.md
```

Still, didn't come up ! 

---

Did I make a mistake anywhere ? Or should I wait for some time for the post to get rendered ?

---

Edit: Such a bummer I am, I looked into issue #27, after posting this issue, but I have also uploaded a key; Did I screw up bigtime  ? My account is **pthk** (I registered on 18th (about a week ago), and ssh key was optional, So, I hadn't uploaded my key, I only had a password)

**Comment by hmngwy** [comment]
@pthk your posts are going through fine, there is a bit of an issue with some updated code. I'll let you know when it is fixed.

@pscollins #28 is in production, we're getting the below which is causing the issue:

```
/var/app/current/lib/file.js:72
    if(this.isTextFile(filename)) {
            ^
TypeError: undefined is not a function
    at module.exports.parse (/var/app/current/lib/file.js:72:13)
    at Query.<anonymous> (/var/app/current/lib/ssh/exec.js:66:23)
    at /var/app/current/node_modules/mongoose/node_modules/kareem/index.js:177:19
    at /var/app/current/node_modules/mongoose/node_modules/kareem/index.js:109:16
    at process._tickDomainCallback (node.js:381:11)
```

Which is very strange, I'm not getting this on a local instance.

**Comment by notpratheek** [comment]
@hmngwy 

Okay ! will wait ! :smile:

**Comment by hmngwy** [comment]
@pthk @pscollins rolled back production code to pre #28. saving should be back up. 

P.S. seeing it in local now.

**Comment by notpratheek** [comment]
@hmngwy 

Yay ! (ﾉ´ヮ´)ﾉ*:･ﾟ✧ (It's getting pushed !)

I have a small question, Could I change the key ? (cause, this one has no passphrase, I want the key to have a passphrase ! )

**Comment by hmngwy** [comment]
@pscollins hotfix pushed (desperate to get latest on production), see if you want to keep these changes, feel free to overwrite on next PR: ffcfd56

`npm test` output same as of #28, 2 failing still because of [this](https://github.com/hmngwy/weblog.sh/pull/28#issuecomment-159856411).

**Comment by notpratheek** [comment]
@hmngwy 

Nevermind about my question on changing the key, I figured it out ! :smile:

**Comment by hmngwy** [comment]
Closing as this was fixed in ffcfd56
